### ISSUE TITLE
check_disk graph are empty

### ISSUE THREAD

**Issue by mbautista** [pnp4nagios/pnp4nagios#62]
**Describe the bug**
Performance graphics for check_disk are empty.

**To Reproduce**
- Configure check_disk on slave nrpe server, for example `command[check_disk]=/usr/lib/nagios/plugins/check_disk -w 20% -c 10% -p /var`
- Configure check_disk on nagios master : 
```
define service{
    use generic-service
    host_name myserver
    servicegroups disk
    service_description Disk
    notifications_enabled 0
    check_interval 720
    check_command check_nrpe!check_disk
}
```
- install pnp4nagios
- pnp4nagios generate performance graphics for all other checks (load, http, mysql, etc.)
- pnp4nagios generates empty graphics pour check_disk, disk name, warning and critical values are displayed in graph legend, but there is no curve

**Expected behavior**
pnp4nagios should display a graphic with disk usage.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - NagiosCore : 4.4.6
 - OS: Debian 11.9
 - pnp4nagios : 0.6.27

**Additional context**
- example of the check_disk output : `DISK OK - free space: /var 350976 MB (73% inode=97%);| /var=127176MB;403014;453391;0;503768`
- npcd service is running, no errors in logs, xml and rrd files seem good, example of xml file : 
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<NAGIOS>
  <DATASOURCE>
    <TEMPLATE>check_nrpe</TEMPLATE>
    <RRDFILE>/usr/local/pnp4nagios/var/log/pnp4nagios/rrd/myserver/Disk.rrd</RRDFILE>
    <RRD_STORAGE_TYPE>SINGLE</RRD_STORAGE_TYPE>
    <RRD_HEARTBEAT>8460</RRD_HEARTBEAT>
    <IS_MULTI>0</IS_MULTI>
    <DS>1</DS>
    <NAME>_var</NAME>
    <LABEL>/var</LABEL>
    <UNIT>MB</UNIT>
    <ACT>127225</ACT>
    <WARN>403014</WARN>
    <WARN_MIN></WARN_MIN>
    <WARN_MAX></WARN_MAX>
    <WARN_RANGE_TYPE></WARN_RANGE_TYPE>
    <CRIT>453391</CRIT>
    <CRIT_MIN></CRIT_MIN>
    <CRIT_MAX></CRIT_MAX>
    <CRIT_RANGE_TYPE></CRIT_RANGE_TYPE>
    <MIN>0</MIN>
    <MAX>503768</MAX>
  </DATASOURCE>
```

**Comment by celane** [comment]
Since the other graphs are working, this sounds like
process_perfdata.pl is having problems getting the
data into the RRD database. 

First, it generally takes more than one perfdata 
update before a graph will be generated. So if you
do check_disk every 30 minutes, it may be over 60
minutes before you get a graph. That's how RRD
generates histograms. 

But if that's not the problem, turn on logging for
process_perfdata. There's a script installed in
the "etc/pnp4nagios" directory SetLogLevels
that you can use for that: ./SetLogLevels debug
Or look for the file "process_perfdata.cfg" and
set LOG_LEVEL=2

then look in the logging directory, perfdata.log
to see where the check_disk data is processed, and
if any error messages are thrown. 

On Thu, 21 Mar 2024 02:49:01 -0700
Mathieu Bautista ***@***.***> wrote:

> **Describe the bug**
> Performance graphics for check_disk are empty.
> 
> **To Reproduce**
> - Configure check_disk on slave nrpe server, for example
> `command[check_disk]=/usr/lib/nagios/plugins/check_disk -w 20% -c 10%
> -p /var`
> - Configure check_disk on nagios master : 
> ```
> define service{
>     use generic-service
>     host_name myserver
>     servicegroups disk
>     service_description Disk
>     notifications_enabled 0
>     check_interval 720
>     check_command check_nrpe!check_disk
> }
> ```
> - install pnp4nagios
> - pnp4nagios generate performance graphics for all other checks
> (load, http, mysql, etc.)
> - pnp4nagios generates empty graphics pour check_disk, disk name,
> warning and critical values are displayed in graph legend, but there
> is no curve
> 
> **Expected behavior**
> pnp4nagios should display a graphic with disk usage.
> 
> **Screenshots**
> N/A
> 
> **Desktop (please complete the following information):**
>  - NagiosCore : 4.4.6
>  - OS: Debian 11.9
>  - pnp4nagios : 0.6.27
> 
> **Additional context**
> - example of the check_disk output : `DISK OK - free space: /var
> 350976 MB (73% inode=97%);| /var=127176MB;403014;453391;0;503768`
> - npcd service is running, no errors in logs, xml and rrd files seem
> good, example of xml file : ```
> <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
> <NAGIOS>
>   <DATASOURCE>
>     <TEMPLATE>check_nrpe</TEMPLATE>
>     <RRDFILE>/usr/local/pnp4nagios/var/log/pnp4nagios/rrd/myserver/Disk.rrd</RRDFILE>
>     <RRD_STORAGE_TYPE>SINGLE</RRD_STORAGE_TYPE>
>     <RRD_HEARTBEAT>8460</RRD_HEARTBEAT>
>     <IS_MULTI>0</IS_MULTI>
>     <DS>1</DS>
>     <NAME>_var</NAME>
>     <LABEL>/var</LABEL>
>     <UNIT>MB</UNIT>
>     <ACT>127225</ACT>
>     <WARN>403014</WARN>
>     <WARN_MIN></WARN_MIN>
>     <WARN_MAX></WARN_MAX>
>     <WARN_RANGE_TYPE></WARN_RANGE_TYPE>
>     <CRIT>453391</CRIT>
>     <CRIT_MIN></CRIT_MIN>
>     <CRIT_MAX></CRIT_MAX>
>     <CRIT_RANGE_TYPE></CRIT_RANGE_TYPE>
>     <MIN>0</MIN>
>     <MAX>503768</MAX>
>   </DATASOURCE>
> ```
> 
> 



-- 
Drexel University  |/             /    ·   ν ·   Chuck Lane   |D
======]－－－>－－－C－－－－－π+－－<                Disque 911  |U
Particle Physics    \             \    ~~ e+~~   215-895-1545 |N
***@***.***                    －​μ+－－+νν                 |E

**Comment by mbautista** [comment]
Hello  @celane 
Thanks for your help :)

I turned the debug log on for perfdata, I did not find errors : 

```
2024-03-22 09:42:58 [1947079] [2] Processing Line 9
2024-03-22 09:42:58 [1947079] [2] Datatype set to 'SERVICEPERFDATA' 
2024-03-22 09:42:58 [1947079] [1] Found Performance Data for myserver / Disk (/var=127373MB;403014;453391;0;503768) 
2024-03-22 09:42:58 [1947079] [2] data2rrd called
2024-03-22 09:42:58 [1947079] [2] RRDs::update /usr/local/pnp4nagios/var/log/pnp4nagios/rrd/myserver/Disk.rrd 1711096960:127373
2024-03-22 09:42:58 [1947079] [2] /usr/local/pnp4nagios/var/log/pnp4nagios/rrd/myserver/Disk.rrd updated
```

The check_disk interval is currently 12h (check_interval 720), but the perfdata is running for many days now I should have a graph no ? Or maybe I should set a shorter interval ? I wanted to avoid useless checks because a disk state rarely changes quickly, but I guess 12h might be too high, maybe 1h ?

The perfdata looks like : `/var=127379MB;403014;453391;0;503768`
I noticed that /var is displayed "_var" in the graphic, and there is no value. Could it be a problem with the "/" ? Or maybe with the unit "MB" ? The warning and the critical values are well displayed in the graph legend.

EDIT : I changed set check_interval from 720 (12h) to 60 (1h), I'll let you know if it solves my problem.
